ssl/quic/quic_impl.c

```diff
@@ -455,21 +522,35 @@ int ossl_quic_get_wpoll_descriptor(QUIC_CONNECTION *qc, BIO_POLL_DESCRIPTOR *des
 }
 
 /* SSL_net_read_desired */
+QUIC_TAKES_LOCK
 int ossl_quic_get_net_read_desired(QUIC_CONNECTION *qc)
 {
+    int ret;
+
+    quic_lock(qc);
+
     if (qc->ch == NULL)
         return 0;
 
-    return ossl_quic_reactor_net_read_desired(ossl_quic_channel_get_reactor(qc->ch));
+    ret = ossl_quic_reactor_net_read_desired(ossl_quic_channel_get_reactor(qc->ch));
+    quic_unlock(qc);
+    return ret;
 }
 
 /* SSL_net_write_desired */
+QUIC_TAKES_LOCK
 int ossl_quic_get_net_write_desired(QUIC_CONNECTION *qc)
 {
+    int ret;
+
+    quic_lock(qc);
+
     if (qc->ch == NULL)
         return 0;
```

Comment:
unlock missing here

Reply:
or consider something like this instead:
```c
int ret = 0;

quic_lock(qc);
if (qc->ch != NULL)
    ret = ...;
quic_unlock(qc);
return ret;
```